drivers/source/SFDP.cpp

```diff
@@ -107,20 +111,69 @@ int sfdp_parse_single_param_header(sfdp_prm_hdr *phdr_ptr, sfdp_hdr_info &hdr_in
         return -1;
     }
 
-    if ((phdr_ptr->PID_LSB == 0) && (sfdp_get_param_id_msb(phdr_ptr->DWORD2) == 0xFF)) {
-        tr_debug("Parameter header: Basic Parameter Header");
-        hdr_info.bptbl.addr = sfdp_get_param_tbl_ptr(phdr_ptr->DWORD2);
-        hdr_info.bptbl.size = std::min((phdr_ptr->P_LEN * 4), SFDP_BASIC_PARAMS_TBL_SIZE);
+    int param_id_msb = sfdp_get_param_id_msb(phdr_ptr->DWORD2);
 
-    } else if ((phdr_ptr->PID_LSB == 0x81) && (sfdp_get_param_id_msb(phdr_ptr->DWORD2) == 0xFF)) {
-        tr_debug("Parameter header: Sector Map Parameter Header");
-        hdr_info.smptbl.addr = sfdp_get_param_tbl_ptr(phdr_ptr->DWORD2);
-        hdr_info.smptbl.size = phdr_ptr->P_LEN * 4;
+    /* MSB JEDEC ID */
+    if (param_id_msb == 0xFF) {
 
-    } else {
-        tr_debug("Parameter header: header vendor specific or unknown. Parameter ID LSB: 0x%" PRIX8 "; MSB: 0x%" PRIX8 "",
+        /* LSB JEDEC ID */
+        switch (phdr_ptr->PID_LSB) {
+            case 0x0:
+                tr_debug("Parameter header: JEDEC Basic Flash - Revision %" PRIX8 ".%" PRIX8 "",
+                         phdr_ptr->P_MAJOR,
+                         phdr_ptr->P_MINOR);
+                hdr_info.bptbl.addr = sfdp_get_param_tbl_ptr(phdr_ptr->DWORD2);
+                hdr_info.bptbl.size = std::min((phdr_ptr->P_LEN * 4), SFDP_BASIC_PARAMS_TBL_SIZE);
+                break;
+            /* TODO: Following headers should be parsed */
```

Comment:
We shouldn't have TODO's in released code....

Reply:
This doesn't look ready for coming into master? Should this really be on a feature branch ?

Reply:
This is a good question @VeijoPesonen @SeppoTakalo why are we not parsing these headers? Do we need to mention somewhere not supported cases?

Reply:
If these TODo's are due to currently unsupported devices or functionality, then we should say something along those lines ie 'This is currently unsupported functionality'. This should also be in our docs somewhere ? 'TODO's ' look unprofessional in released code....

Reply:
> We shouldn't have TODO's in released code....

Personally I would prefer to see if something is not implemented. I wouldn't want to give a wrong impression that SFDP database parsing takes into consideration all possible combinations as long as it's not true.

Reply:
> If these TODo's are due to currently unsupported devices or functionality, then we should say something along those lines ie 'This is currently unsupported functionality'. This should also be in our docs somewhere ? 'TODO's ' look unprofessional in released code....

TODO -> UNSUPPORTED, agreed.

Reply:
> This doesn't look ready for coming into master? Should this really be on a feature branch ?

The plan is not to implement parsing of all possible headers so I don't see a reason to introduce a feature branch.